src/Form/MailTemplateCustomDataType.php

```diff
@@ -28,13 +29,23 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
         if (null !== $options['template']) {
             $customFields = $this->mailTemplateHelper->getCustomFields($options['template']);
 
-            foreach ($customFields as $key => $label) {
-                $builder->add($key, TextType::class, [
-                    'label' => $label,
-                    'mapped' => false,
-                    'data' => isset($options['data'][$key]) && !empty($options['data'][$key]) ? $options['data'][$key] : '',
-                    'required' => false,
-                ]);
+            foreach ($customFields as $key => $value) {
+                if ('textArea' === $value['type']) {
+                    $builder->add($key, TextareaType::class, [
+                        'label' => $value['label'],
+                        'mapped' => false,
+                        'data' => isset($options['data'][$key]) && !empty($options['data'][$key]) ? $options['data'][$key] : '',
+                        'required' => false,
+                        'attr' => ['rows' => 5],
+                    ]);
+                } elseif ('text' === $value['type']) {
+                    $builder->add($key, TextType::class, [
+                        'label' => $value['label'],
+                        'mapped' => false,
+                        'data' => isset($options['data'][$key]) && !empty($options['data'][$key]) ? $options['data'][$key] : '',
+                        'required' => false,
+                    ]);
```

Comment:
Much of this code is the same Consider rewriting to something like

```php
$type  = TextType::class;
$options = [
  'label' => $value['label'],
  'mapped' => false,
  …
];

if ('textArea' === $value['type']) {
  $options['attr'] = …;
}

$builder->add($key, $type, $options);
```

Reply:
Updated in 81896df1e4a3900c171dacc9dbf863168bf41171.